src/main/java/org/terracotta/statistics/registry/StatisticRegistry.java

```diff
@@ -106,20 +107,15 @@ public StatisticRegistry(Object contextObject, Supplier<Long> timeSource) {
 
   @SuppressWarnings("unchecked")
   public Map<String, Statistic<? extends Serializable>> queryStatistics(long sinceMillis) {
-    Map<String, Statistic<? extends Serializable>> stats = new HashMap<>(statistics.size());
-    statistics.forEach((name, stat) -> stats.put(name, new Statistic<>(stat.type(), filter(stat.type(), stat.history(sinceMillis)))));
-    return stats;
+    long now = timeSource.getAsLong();
+    return statistics.entrySet().stream().collect(toMap(Map.Entry::getKey, e -> statistic(e.getValue(), now, sinceMillis)));
```

Comment:
Not totally related to this PR but it feels weird that the parameters are `now`, `since`. Because the classical order would be `from`, `to`. The fact that `to` is actually `now` doesn't matter.

Reply:
it is related: this is to avoid having to do several consecutive calls to `System.currentTimeMillis`. I just refactored the static method to be in a better place.

Reply:
But why can't we have `statistic(e.getValue(), sinceMillis, now)`?

Reply:
I also renamed the parameters but I disagree with from/to because these parameters are not defining a range: `now` is the time when the user called the query method and `since` is the time from when the user wants the stats. So the sample history returned could have stats containing timestamps >= now and this is legal.

Reply:
I my proposal doesn't make sense, don't do it :-)